hayeon/programmers-1/성격-유형-검사하기.js

```diff
@@ -0,0 +1,22 @@
+function solution(survey, choices) {
+  const MBTI = {};
+  const types = ["RT", "CF", "JM", "AN"];
+
+  types.forEach((type) => type.split("").forEach((char) => (MBTI[char] = 0)));
+
+  choices.forEach((choice, index) => {
+    const [disagree, agree] = survey[index];
+
+    MBTI[choice > 4 ? agree : disagree] += Math.abs(choice - 4);
+  });
+
+  return types.map(([a, b]) => (MBTI[b] > MBTI[a] ? b : a)).join("");
+}
```

Comment:
```suggestion
function solution(survey, choices) {
  const types = ["RT", "CF", "JM", "AN"];
  const MBTI = {
    R: 0,
    T: 0,
    C: 0,
    F: 0,
    J: 0,
    M: 0,
    A: 0,
    N: 0,
  };

  choices.forEach((choice, index) => {
    const [disagree, agree] = survey[index];
    const choiceType = choice > 4 ? agree : disagree

    MBTI[choiceType] += Math.abs(choice - 4);
  });

  return types.map(([a, b]) => (MBTI[b] > MBTI[a] ? b : a)).join("");
}
```
- 풀이 잘봤습니다 🙇‍♂️
- `types.forEach((type) => type.split("").forEach((char) => (MBTI[char] = 0)));` 를 통해 객체로 초기값을 설정해주셨는데, 해당문제는 유형이 정해져있으니 그냥 처음에 선언해주셔도 좋을 것 같아요
- 객체 조회에 `[choice > 4 ? agree : disagree]` 와 같은 부분 변수로 빼서 사용해주세요 ~!

Reply:
넵 알겠습니다!! 이문제 너무 어려워서..... ㅜㅜ 여러개 참고해봤는데 풀이도 어렵더라구욥....윽..